### PR TITLE
Allow PROTOBUF_LIBRARY option via cmake command

### PR DESCRIPTION
**Pull request by nalinigans**
An alternative to using `PROTOBUF_LIBRARY` would be to use `CMAKE_PREFIX_PATH` instead. Also, did some minor cleanup of CMakeLists.txt. Note that `OPENSSL_PREFIX_DIR` has been removed as `OPENSSL_ROOT_DIR` can be passed either as a cmake option or setup as an environment variable.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,54 +25,65 @@
...
 set(GENOMICSDB_VERSION "${GENOMICSDB_RELEASE_VERSION}-${GIT_COMMIT_HASH}" CACHE STRING "GenomicsDB full version string")
+
 set(DISABLE_MPI False CACHE BOOL "Disable use of any MPI compiler/libraries")
 set(DISABLE_OPENMP False CACHE BOOL "Disable OpenMP")
+
 set(BUILD_DISTRIBUTABLE_LIBRARY False CACHE BOOL "Build the TileDB/GenomicsDB library with minimal runtime dependencies")
 set(BUILD_JAVA False CACHE BOOL "Build Java/JNI interface for combined VCF records")
+
 set(HTSLIB_INSTALL_DIR "" CACHE PATH "Path to htslib install directory")
-set(OPENSSL_PREFIX_DIR "" CACHE PATH "OpenSSL installation prefix")
```
Comment (kgururaj) [inline CMakeLists.txt:58]: Why delete this?

File: CMakeLists.txt [hunk 2]
```diff
@@ -26,54 +25,65 @@
...
 set(MAVEN_QUIET False CACHE BOOL "Do not print mvn messages")
 set(GENOMICSDB_SPARK_PROFILE "" CACHE STRING "Profile to choose Apache Spark version") # used in Maven builds
-set(IPPROOT "" CACHE PATH "Path to IPP libraries - used when optimized zlib is used")
-set(GENERATE_PROTOBUF_FILES_IN_BUILD_DIR True CACHE BOOL "Generate all protobuf files in build directory")
+
+set(LIBDBI_DIR "" CACHE PATH "Path to libdbi install directory")
+
+set(IPPROOT "" CACHE PATH "Path to IPP libraries - used when intel optimized zlib is required")
+
+set(LIBDBI_DIR "" CACHE PATH "Path to libdbi install directory")
```
Comment (kgururaj) [inline CMakeLists.txt:84]: Duplicate?

File: cmake/Modules/FindTileDB.cmake [hunk 3]
```diff
@@ -48,10 +48,6 @@ set(TILEDB_DISABLE_TESTING True)
 #Zlib
 find_package(ZLIB REQUIRED)
 
-#OpenSSL
-if(OPENSSL_PREFIX_DIR AND NOT OPENSSL_ROOT_DIR)
-    set(OPENSSL_ROOT_DIR "${OPENSSL_PREFIX_DIR}")
-endif()
```
[the remark this answers is not in the document]
Comment (nalinigans) [reply cmake/Modules/FindTileDB.cmake:54]: `OPENSSL_PREFIX_DIR` does not seem to be used at least from 3.6 onwards - see https://cmake.org/cmake/help/v3.6/module/FindOpenSSL.html. As `OPENSSL_ROOT_DIR` can pretty much do what `OPENSSL_PREFIX_DIR` is intended for, do you think we need to have both?

File: CMakeLists.txt (hunk 1 above)
Comment (nalinigans) [reply CMakeLists.txt:58]: See my comment above. I can reinstate if you want to.

File: CMakeLists.txt (hunk 2 above)
Comment (nalinigans) [reply CMakeLists.txt:82]: Thanks for catching. Removed the duplicate.

**Review by kgururaj — APPROVED** [review]
Looks good

**Pull request merged**